app/filters/document_provider.rb

```diff
@@ -0,0 +1,62 @@
+class DocumentProvider
+  attr_reader :current_site, :document_type, :keyword, :filters
+
+  BLOCKS_TO_SEARCH = %w[content overview]
+
+  def initialize(current_site, document_type, keyword, filters)
+    @current_site = current_site
+    @document_type = document_type
+    @keyword = keyword
+    @filters = filters
```

Comment:
I think this PR is in a good state. All the tests now are clearer to understand each context and the data involved. My only question is related to the whole approach. Do you have a future plan on how the pagination will work? Will it be one more argument to this DocumentProvider or will be handled only on the controller?

Reply:
By the way I know that the pagination will not be addressed on this PR, but would be good to start to think about it since it affects the current solution.